src/Integration/Notifications/SonarQubeNotificationService.cs

```diff
@@ -86,48 +88,53 @@ public void Stop()
 
         private async Task UpdateEvents()
         {
-            if (!sonarQubeService.IsConnected)
+            if (!sonarQubeService.IsConnected ||
+                !Model.AreNotificationsEnabled)
             {
                 return;
             }
 
-            var events = await GetNotificationEvents();
-            if (events == null)
+            try
             {
-                // Notifications are not supported on SonarQube or an exception was thrown
-                Stop();
-                return;
+                var events = await GetNotificationEvents();
+                if (events == null)
+                {
+                    // Notifications are not supported on SonarQube
+                    Stop();
+                    return;
+                }
+
+                if (!isFirstServerQuery)
+                {
+                    if (events.Count > 0)
+                    {
+                        lastCheckDate = events.Max(ev => ev.Date);
+                    }
+                    Model.SetNotificationEvents(events);
+                }
+
+                Model.IsIconVisible = true;
             }
-
-            if (events.Count > 0)
+            catch (Exception ex)
             {
-                lastCheckDate = events.Max(ev => ev.Date);
+                sonarLintOutput.Write($"Failed to fetch notifications : {ex.Message}");
             }
 
-            Model.IsIconVisible = true;
-            Model.SetNotificationEvents(events);
+            isFirstServerQuery = false;
         }
 
         private async void OnTimerElapsed(object sender, EventArgs e)
         {
-            Debug.Assert(this.cancellation != null, "Cancellation token should not be null if the timer is active - check StartAsync has been called");
+            Debug.Assert(cancellation != null,
+                "Cancellation token should not be null if the timer is active - check StartAsync has been called");
 
             await UpdateEvents();
         }
 
         private async Task<IList<SonarQubeNotification>> GetNotificationEvents()
         {
-            try
-            {
-                return await sonarQubeService.GetNotificationEventsAsync(projectKey,
-                    lastCheckDate, cancellation.Token);
-            }
-            catch (Exception ex)
-            {
-                sonarLintOutput.Write($"Failed to fetch notifications : {ex.Message}");
-
-                return null;
-            }
+            return await sonarQubeService.GetNotificationEventsAsync(projectKey,
```

Comment:
Perhaps you should get rid of the GetNotificationEvents() method.

Reply:
Agreed with @valhristov the method is only doing a one line call with a really close name so I would remove this method.